plugins/headlamp-plugin/bin/headlamp-plugin.js

```diff
@@ -316,6 +348,205 @@ function format(packageFolder) {
+      encoding: 'utf8',
+    });
+    // console.log(result);
+  } catch (error) {
+    // npm outdated exit codes 1 when something is not up to date.
+    result = error.stdout.toString();
+  }
+  return JSON.parse(result);
+}
+
+/**
+ * Upgrade package automatically, updating headlamp-plugin version.
+ *
+ * In the future this could be used for other upgrade tasks.
+ *
+ * @param packageFolder {string} - folder where the package, or folder of packages is.
+ * @returns {0 | 1} Exit code, where 0 is success, 1 is failure.
+ */
+function upgrade(packageFolder, skipPackageUpdates) {
+  /**
+   * Files from the template might not be there.
+   *
+   * Either because they created the package themselves,
+   *   or used an old version headlamp-plugin create.
+   *
+   * Assumes we are in the package folder.
+   */
+  function addMissingTemplateFiles() {
+    const missingFiles = [
+      path.join('src', 'headlamp-plugin.d.ts'),
+      'tsconfig.json',
+      'jsconfig.json',
+    ];
+    const templateFolder = path.resolve(__dirname, '..', 'template');
+
+    missingFiles.forEach(pathToCheck => {
+      const from = path.join(templateFolder, pathToCheck);
+      const to = path.join('.', pathToCheck);
+
+      // only copy it if it doesn't exist
+      if (!fs.existsSync(to)) {
+        console.log(`Adding missing file: "${to}"`);
+        fs.copyFileSync(from, to);
+      }
+    });
+  }
+
+  /**
+   * Adds missing config into package.json
+   */
+  function addMissingConfiguration() {
+    const templateFolder = path.resolve(__dirname, '..', 'template');
+    const packageJsonPath = path.join('.', 'package.json');
+    const templatePackageJson = JSON.parse(
+      fs.readFileSync(path.join(templateFolder, 'package.json'), 'utf8')
+    );
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    let changed = false;
+
+    if (packageJson?.scripts?.tsc !== templatePackageJson['scripts']['tsc']) {
+      packageJson['scripts']['tsc'] = templatePackageJson['scripts']['tsc'];
+      changed = true;
+      console.log(`Updated package.json field "scripts.tsc"`);
+    }
+
+    const checkKeys = ['eslintConfig', 'prettier'];
+    checkKeys.forEach(key => {
+      if (JSON.stringify(packageJson[key]) !== JSON.stringify(templatePackageJson[key])) {
+        packageJson[key] = templatePackageJson[key];
+        changed = true;
+        console.log(`Updated package.json field "${key}"`);
+      }
+    });
+
+    if (changed) {
+      fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, '  ') + '\n');
+    }
+  }
+
+  function upgradeHeadlampPlugin(skipPackageUpdates) {
+    if (skipPackageUpdates === true) {
+      console.log('Skipping package updates.');
+      return true;
+    }
+
+    const outDated = getNpmOutdated();
+    if ('@kinvolk/headlamp-plugin' in outDated) {
+      // Upgrade the @kinvolk/headlamp-plugin
+
+      const cmd = 'npm install @kinvolk/headlamp-plugin@latest --save';
```

Comment:
Won't this get the version to be `latest` in the package.json? Probably doesn't matter with package-lock.json, but I wonder if this is the case.

Reply:
I checked this. It sets the version to the latest version, not to "latest". Now it sets it to "^0.5.1" for example.